experimental/conn/analog/analog.go

```diff
@@ -2,34 +2,46 @@
 // Use of this source code is governed under the Apache License, Version 2.0
 // that can be found in the LICENSE file.
 
-// Package analog defines analog pins, both DAC and ADC.
+// Package analog defines analog pins, both digital to analog converter (DAC)
+// and analog to digital converter (ADC).
 package analog
 
 import (
-	"errors"
-
+	"periph.io/x/periph/conn/physic"
 	"periph.io/x/periph/conn/pin"
 )
 
-// ADC is an analog-to-digital-conversion input.
-type ADC interface {
+// Reading is the result of PinADC.Read().
+type Reading struct {
+	// V is the interpreted electrical level.
+	V physic.ElectricPotential
+	// Raw is the raw measurement.
+	Raw int32
+}
+
+// PinADC is an analog-to-digital-conversion input.
+type PinADC interface {
 	pin.Pin
 	// Range returns the maximum supported range [min, max] of the values.
-	Range() (int32, int32)
+	Range() (Reading, Reading)
 	// Read returns the current pin level.
-	Read() int32
+	Read() (Reading, error)
+	// ReadContinuous opens a channel and reads continuously until Halt() is
+	// called.
+	ReadContinuous() <-chan Reading
 }
 
-// DAC is an digital-to-analog-conversion output.
-type DAC interface {
+// PinDAC is an digital-to-analog-conversion output.
+type PinDAC interface {
 	pin.Pin
 	// Range returns the maximum supported range [min, max] of the values.
 	Range() (int32, int32)
 	// Out sets an analog output value.
-	DAC(v int32)
+	Out(v int32) error
```

Comment:
I do not see much value in the DAC to accept an ElectricPotential value.
The challenge here is that this makes DAC and ADC not symmetrical. This is especially visible with Range(). This means that a pin cannot implement both interfaces simultaneously.
In practice I wouldn't expect an analog pin to be directional; unlike digital pins (gpio) which are usually bidirectional, analog pins rarely are. That said I would prefer to not have this issue.

One option would be to have Range() here to return Reading, even if it may not make sense, I don't know.